src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -38,7 +39,20 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 			return await yupResolver(hardValidationSchema)(data, context, options);
 		},
 	});
-	const { reset, watch, handleSubmit: reactFormHookSubmit, getValues } = formMethods;
+
+	const {
+		reset,
+		watch,
+		handleSubmit: reactFormHookSubmit,
+		getValues,
+		setError,
+		control,
+		formState: { errors },
+		clearErrors,
+	} = formMethods;
+
+	const formValues = useWatch({ control });
```

Comment:
there is a need to disable `useWatch` if there is nothing to watch so it is not watching unnecessarily

actually the previous implementation to watch only if there are api errors is fine too

Reply:
noticed that `watch` only unsubscribes upon unmount, hence decided to stick with `useWatch` to disable it dynamically